### ISSUE TITLE
Lambda with error branch and item.json property

### ISSUE THREAD

**Issue by barn4k** [n8n-io/n8n#10260]
### Bug Description

If the lambda node is set to have an error output branch, then the .item.json property won't work for any nodes before lambda. It doesn't matter of which branch has been chosen.
![image](https://github.com/user-attachments/assets/7607ee73-ac28-4da0-b7c2-55eab0343627)

![image](https://github.com/user-attachments/assets/e9026c32-b393-4dca-bb28-847377487d24)

### To Reproduce

Lambda example:
python 3.6+ (or anything you like)
```python
def lambda_handler(event, context):
    if event.get("action", False):
        if event['action'] == 'help':
            return {"status": "ok"}
        else:
            raise ValueError("incorrect action, valid is 'help'")
    else:
        raise ValueError("no 'action' is specified")
```

Workflow example
```json
{
  "meta": {
    "templateCredsSetupCompleted": true,
    "instanceId": "3544babe0875735a9326c77025b337da4da0168a277093d955bd6e325182435a"
  },
  "nodes": [
    {
      "parameters": {},
      "id": "4b14ce5d-3c39-4479-a08b-489ec82bc249",
      "name": "When clicking ‘Test workflow’",
      "type": "n8n-nodes-base.manualTrigger",
      "typeVersion": 1,
      "position": [
        1560,
        540
      ]
    },
    {
      "parameters": {
        "function": "arn:aws:lambda:us-west-2:989621091465:function:soc-prd-GraphAPI",
        "payload": "={{ JSON.stringify($json) }}"
      },
      "id": "ccd5fc44-761c-4a53-8b3a-24d7b1f7107d",
      "name": "AWS Lambda",
      "type": "n8n-nodes-base.awsLambda",
      "typeVersion": 1,
      "position": [
        2040,
        540
      ],
      "credentials": {
        "aws": {
          "id": "4nwRquJ0C2ojN4SB",
          "name": "AWS PRD [Do not Edit]"
        }
      },
      "onError": "continueErrorOutput"
    },
    {
      "parameters": {},
      "id": "b5d757cd-3740-4707-89bd-4f7d836b3258",
      "name": "success branch",
      "type": "n8n-nodes-base.noOp",
      "typeVersion": 1,
      "position": [
        2320,
        420
      ]
    },
    {
      "parameters": {},
      "id": "a469ee08-7c57-43fc-9e2f-19e160d2e544",
      "name": "error branch",
      "type": "n8n-nodes-base.noOp",
      "typeVersion": 1,
      "position": [
        2320,
        620
      ]
    },
    {
      "parameters": {
        "assignments": {
          "assignments": [
            {
              "id": "017a2632-ddf3-4c3f-ad12-7f913d70a66e",
              "name": "status",
              "value": "failed",
              "type": "string"
            },
            {
              "id": "f56637b3-b108-4649-bbcb-982178196701",
              "name": "payload",
              "value": "={{ $('set payload').item.json }}",
              "type": "object"
            }
          ]
        },
        "options": {}
      },
      "id": "7e4f9744-5895-4834-aded-8a8bf18095d2",
      "name": "set fail",
      "type": "n8n-nodes-base.set",
      "typeVersion": 3.3,
      "position": [
        2500,
        620
      ]
    },
    {
      "parameters": {},
      "id": "817b1c70-6d40-442c-8253-27723774d6f8",
      "name": "End",
      "type": "n8n-nodes-base.noOp",
      "typeVersion": 1,
      "position": [
        2800,
        520
      ]
    },
    {
      "parameters": {
        "assignments": {
          "assignments": [
            {
              "id": "017a2632-ddf3-4c3f-ad12-7f913d70a66e",
              "name": "status",
              "value": "success",
              "type": "string"
            },
            {
              "id": "f56637b3-b108-4649-bbcb-982178196701",
              "name": "payload",
              "value": "={{ $('set payload').item.json }}",
              "type": "object"
            }
          ]
        },
        "options": {}
      },
      "id": "04aa5006-ce6c-4b81-92ba-42b68459117d",
      "name": "set succ",
      "type": "n8n-nodes-base.set",
      "typeVersion": 3.3,
      "position": [
        2500,
        420
      ]
    },
    {
      "parameters": {
        "jsCode": "return [\n  {\"action\": \"help\"},\n  {\"action\": \"help3123\"},\n]"
      },
      "id": "07ab57d3-07b2-4e0e-a552-b3f572e2c11c",
      "name": "set payload",
      "type": "n8n-nodes-base.code",
      "typeVersion": 2,
      "position": [
        1780,
        540
      ]
    }
  ],
  "connections": {
    "When clicking ‘Test workflow’": {
      "main": [
        [
          {
            "node": "set payload",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "AWS Lambda": {
      "main": [
        [
          {
            "node": "success branch",
            "type": "main",
            "index": 0
          }
        ],
        [
          {
            "node": "error branch",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "success branch": {
      "main": [
        [
          {
            "node": "set succ",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "error branch": {
      "main": [
        [
          {
            "node": "set fail",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "set fail": {
      "main": [
        [
          {
            "node": "End",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "set succ": {
      "main": [
        [
          {
            "node": "End",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "set payload": {
      "main": [
        [
          {
            "node": "AWS Lambda",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  },
  "pinData": {}
}
```

The issue won't happen if the Lambda's output will be pinned.

### Expected behavior

The item.json property works as expected

### Operating System

docker

### n8n Version

1.45.1

### Node.js Version

docker

### Database

PostgreSQL

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @barn4k,

Thanks for the report, I have put in a fix which be available in next weeks release. The issue with this one is the node didn't correctly support Paired Items which was highlighted more when using the error branch.

**Comment by janober** [comment]
Fix got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)
